### PR TITLE
Updated gRPC names & functionality to match swagger format

### PR DESCRIPTION
**Pull request by nativraz1**
- Better names for gRPC functions & fields
- Added skip & Limit in [GetAppointment](https://github.com/TekClinic/Appointments-MicroService/tree/feature/swagger-format#getappointment)
- Fixed problems in [Current Behavior](https://github.com/TekClinic/Appointments-MicroService/tree/main#Current-Behavior): Upon creating/updating an appointment, only patient_id is optional, and patient_id and doctor_id values are made sure to be non-negative.

All the relevant documentation can be found in [README.md](https://github.com/TekClinic/Appointments-MicroService/tree/feature/swagger-format#Appointments-MicroService).

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -152,18 +167,28 @@ func (server appointmentsServer) GetAppointments(ctx context.Context,
...
 	}
 
 	// Fetch appointments based on filters
 	baseQuery := server.db.NewSelect().Model((*Appointment)(nil))
 
 	// Filter by date range
-	baseQuery = baseQuery.Where("start_time >= ?", date).Where("end_time <= ?", date.AddDate(0, 0, 1))
+	dateStr := req.GetDate()
+	if dateStr != "" {
+		date, dateErr := time.Parse("2006-01-02", dateStr)
```
Comment (vMaroon) [inline server/server.go]: Make date format a const.

I recall seeing this somewhere else, note that if different components need to be aligned on this point, it is best to extract the formatting (and other consts) to a new data-types repo or whatever meaningful name you select. Alternatively, if they should be aligned, then the minimum would be for one to import it from the other.

Opening an issue on the matter is sufficient for now. Let me know what you choose.

**Review by vMaroon — APPROVED** [review]
Looks good, one comment.

File: server/server.go (hunk 1 above)
Comment (nativraz1) [reply server/server.go]: I added the relevant const, where should I open an issue regarding the data-types repo?

Comment (vMaroon) [reply server/server.go]: @nativraz1 backlog

Comment (nativraz1) [reply server/server.go]: done. (https://github.com/TekClinic/backlog/issues/36)

**Review by vMaroon — APPROVED** [review]

**Pull request merged**